### DISCUSSION TITLE
[Feature] Mark automatically recognized faces as unconfirmed, allow user to confirm

### DISCUSSION THREAD

**Discussion by Germs2004** [immich-app/immich#12931] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When the AI automatically recognizes a face as belonging to a certain Person, Immich should flag that face as "unconfirmed".  Then, it should allow the user to click a Person, filter to show only the "unconfirmed" ones for that Person, and then manually select one or more of them to mark them as "confirmed".

Later, if the AI attempts to refresh its guesses about which face belongs to which person, the AI can overwrite its previous unconfirmed guesses, but it should trust any faces marked confirmed and leave those alone.

Ideally, the user should eventually go through all the unconfirmed faces, fix them if necessary, and mark them all as confirmed.  It's the only way to be sure all the faces are recognized correctly without losing all your manual corrections if you let the AI refresh its guesses.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by cratoo** [comment]
Can only second this. Back at the times Picasa did also work like this. In my opinion this could also be optional.
With lots of group pictures of school classes, any AI still seems to struggle sometimes to differentiate especially young kids. To avoid the hassle of manually sorting out the mistakes, the need to confirm would help a lot.

**Comment by marvil6** [comment]
I thought this was how it worked already, at least under the hood — if not, we'd love to see this.

**Comment by urbanrider** [comment]
This would be a great feature

**Comment by Jero1ste** [comment]
Related to feature request #31310, which addresses this same core problem from the backend/job side.

**Real-world scenario:**
I have a 200,000+ photo library spanning 25 years, including two brothers with a ~1.5 years age difference. At young ages, their facial embeddings are very similar, causing the AI to frequently cluster them as the same person.

After manually correcting hundreds of misassignments and lowering the recognition distance threshold from 0.45 to 0.40, there is currently no way to apply the new setting to already-clustered faces without a full **Reset** — which destroys all manual corrections.

The confirmation workflow proposed here, combined with a **"Re-cluster unconfirmed/auto-assigned faces only"** job, would solve this completely. Both features complement each other perfectly:
1. The UI workflow (this issue) lets users confirm/verify faces.
2. The re-cluster job re-evaluates unconfirmed/auto-assigned assets when settings change, while strictly preserving confirmed/manually assigned faces.

**Comment by Jero1ste** [comment]
With v3.2.0 and the introduction of Cluster Groups / Cross-user 
clustering, this request has become more urgent than ever.

The release notes explicitly state:
> "As of now, this requires resetting facial recognition for all 
> users in the group [...] names and birth dates will be lost."

This creates a recurring dilemma with every Immich release that 
touches clustering:
- **Adopt new features** → lose all manually corrected face 
  assignments
- **Skip new features** → miss improvements

For users managing family archives — especially with similar-
looking siblings or toddlers — this is becoming a growing barrier 
to adoption, not just a nice-to-have.

A persistent `isManual` / `isConfirmed` flag, treating manually 
verified faces as immutable anchors during any re-cluster or 
cross-user merge, would eliminate this friction entirely. 
Combined with a "Re-cluster unconfirmed faces only" job 
(see #31310), users could:

✅ Safely adopt Cluster Groups  
✅ Re-cluster without risking manual corrections  
✅ Future-proof their work against model and feature updates  

This pattern will only become more painful with each release.

**Reply by xamino** [reply]
To add my support – I've spent time with non-technical users manually verifying faces on old scanned images to tag our ancestors correctly, and this is work I'd love to prevent having to re-do, especially since it involves sitting down again and redoing already done work with my users. This also includes filling out things like birthdays, when known, etc.

**Reply by urbanrider** [reply]
Agreed. Also the longer Immich is around more people will have spent more and more time curating their names, faces, etc. If someone has many years of work in tagging and an update clears all of that, people wont be happy. It would be nice to know that the time you spent on assigning names and faces is stored in a way that will persist.

**Reply by CarrotManMatt** [reply]
I'd love this also. Even right now I'm a bit confused as to what would get reset and what wouldn't both currently and under this proposal. Would whole images be tagged as "manually edited" or just individual faces? Is it individual faces in a photo that would be affected by a reset, or all of a person's faces and date of birth information (as the release docs suggest). Apologies for my confusion, if anyone could clear it up I'd very much appreciate that!

**Reply by Jero1ste** [reply]
Great question! As Immich currently works with a full facial recognition reset (via Administration > Facial Recognition), you essentially wipe the entire cluster database. That means all people, links, and date of birth information are lost; Immich then has to rescan and recluster all photos from scratch.

What this proposal aims to achieve is granularity at the individual face level (bounding box), not at the photo level. Once you manually assign, rename, or confirm a face, that specific face receives a status flag (such as 'isManual' or 'confirmed'). During any reindexing or reclustering, those manually confirmed links remain untouched (locked), while only the unconfirmed/automatic detections are recalculated. You would no longer lose your manual work or any person metadata.

**Reply by marvil6** [reply]
I had shot myself in the foot before with this. After manually tagging and clustering faces manually for months, I wanted to update the facial recognition settings, so I cleared the matches —thinking, "obviously, this won't reset the matches _I made manually_ — only the machine-matched faces"— so I completely wiped all of my work. I haven't tagged any face at all since then, and it's been years :(

**Reply by Jero1ste** [reply]
@marvil6 That's exactly the painful experience this proposal 
aims to prevent. The assumption that "manual = safe" is 
completely reasonable — it's how most software works. 
The fact that Immich doesn't distinguish between manual and 
automatic assignments is the core of the problem.

Hope we'll see this fixed soon. The support in this thread 
shows how many people have been caught by the same trap.

**Reply by ffchung** [reply]
Why not just simply make all named person as it own cluster group at start (Acc A - Tony -> Group 1, Acc B - Tony -> Group 2) . and reset all other unname. And let us to merge it ourselves .
